libbf.c

```diff
@@ -1710,6 +1710,11 @@ static int __bf_div(bf_t *r, const bf_t *a, const bf_t *b, limb_t prec,
         slimb_t d;
 
         na = n + nb;
+
+        if (na >= (SIZE_MAX / sizeof(limb_t)) - 1) && !defined(_WIN32) {
```

Comment:
I think this check should be surrounded by some ifdef sizeof (limb_t) > sizeof(uint64_t) instead